applications/RANSApplication/custom_conditions/vms_monolithic_k_based_wall_condition.cpp

```diff
@@ -0,0 +1,200 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <functional>
+#include <iostream>
+#include <limits>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/variables.h"
+
+// Application includes
+#include "custom_utilities/rans_calculation_utilities.h"
+#include "rans_application_variables.h"
+
+// Include base h
+#include "vms_monolithic_k_based_wall_condition.h"
+
+namespace Kratos
+{
+template <unsigned int TDim, unsigned int TNumNodes>
+int VMSMonolithicKBasedWallCondition<TDim, TNumNodes>::Check(
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    int check = BaseType::Check(rCurrentProcessInfo);
+
+    const auto& r_geometry = this->GetGeometry();
+
+    for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+        const auto& r_node = r_geometry[i_node];
+
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(TURBULENT_KINETIC_ENERGY, r_node);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DENSITY, r_node);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VELOCITY, r_node);
+    }
+
+    return check;
+
+    KRATOS_CATCH("");
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void VMSMonolithicKBasedWallCondition<TDim, TNumNodes>::Initialize()
+{
+    KRATOS_TRY;
+
+    if (RansCalculationUtilities::IsWallFunctionActive(*this)) {
+        const array_1d<double, 3>& r_normal = this->GetValue(NORMAL);
+        KRATOS_ERROR_IF(norm_2(r_normal) == 0.0)
+            << "NORMAL must be calculated before using this " << this->Info() << "\n";
+
+        KRATOS_ERROR_IF(this->GetValue(NEIGHBOUR_ELEMENTS).size() == 0)
+            << this->Info() << " cannot find parent element\n";
```

Comment:
Shouldn't be these checks performed in the `Check` function?

Reply:
The normals are calculated somewhere after check and Initialize, so I have to check them here :/